### PR TITLE
Update to docs

### PR DESCRIPTION
**Pull request by GavT**
### Why is this change needed?

- more detail added to docs for SC deployment

### What changes were made as part of this PR:

- more accurate description on using remix to deply SC

### CODE REVIEW

File: docs/testnet/deploying-a-smart-contract.md [hunk 1]
```diff
@@ -2,23 +2,35 @@
...
-* MetaMask installed in your browser.
-* A local copy of the Obscuro MetaMask wallet extension downloaded from the [Obscuro repo on GitHub](https://github.com/obscuronet/go-obscuro/tree/main/tools/walletextension).
+* [MetaMask](https://metamask.io/) wallet installed in your browser.
+* A local copy of the Obscuro MetaMask wallet extension downloaded from the [releases area](https://github.com/obscuronet/go-obscuro/releases) Obscuro repository on GitHub.
 
 ## Prepare Your MetaMask Wallet for Obscuro Testnet
 An essential part of how Obscuro provides full privacy is the encryption of communication between an Obscuro application and Obscuro nodes on the network.
 
-Follow the steps to config the MetaMask wallet extension [here](https://docs.obscu.ro/testnet/wallet-extension.html) then return to this page.
+Follow the steps to config the MetaMask wallet extension [here](wallet-extension.md) then return to this page. If you do not have the Obscuro wallet extension running MetaMask will not be able to communicate with the Obscuro Testnet.
```
Comment (joeldudley) [inline docs/testnet/deploying-a-smart-contract.md:11]: You need to both configure and run it (or you could add running it to the steps below).

File: docs/testnet/deploying-a-smart-contract.md [hunk 2]
```diff
@@ -2,23 +2,35 @@
...
+* A local copy of the Obscuro MetaMask wallet extension downloaded from the [releases area](https://github.com/obscuronet/go-obscuro/releases) of the Obscuro repository on GitHub.
 
 ## Prepare Your MetaMask Wallet for Obscuro Testnet
 An essential part of how Obscuro provides full privacy is the encryption of communication between an Obscuro application and Obscuro nodes on the network.
 
-Follow the steps to config the MetaMask wallet extension [here](https://docs.obscu.ro/testnet/wallet-extension.html) then return to this page.
+Follow the steps to config the MetaMask wallet extension [here](wallet-extension.md) then return to this page. If you do not have the Obscuro wallet extension running MetaMask will not be able to communicate with the Obscuro Testnet. 
 
-Now your wallet is configured to make your encrypted traffic viewable you can go ahead and deploy your smart contract to the Obscuro Testnet.
+Now your wallet is configured for the Obscuro Testnet which makes your encrypted traffic viewable. You can now go ahead and deploy your smart contract to the Obscuro Testnet.
```
Comment (joeldudley) [inline docs/testnet/deploying-a-smart-contract.md:13]: "which makes your encrypted traffic viewable" - I think this makes it sound like everyone can see it. Maybe, "which allows you to view encrypted traffic for your wallet only"?

File: docs/testnet/deploying-a-smart-contract.md [hunk 3]
```diff
@@ -2,23 +2,35 @@
...
 
 ## Deploy Your Smart Contract Using an IDE
-1. Open your favourite Solidity-compatible Integrated Development Environment. The steps below will assume you are using [Remix](https://github.com/ethereum/remix-ide).
+1. Browse to the popular Solidity-compatible Integrated Development Environment called [Remix](https://remix.ethereum.org/).
 
-1. Open your solidity smart contract in Remix.
+1. Load your solidity smart contract into the Remix workspace.
```
Comment (joeldudley) [inline docs/testnet/deploying-a-smart-contract.md:18]: Maybe link to some Remix docs for this and the following section.

File: docs/testnet/deploying-a-smart-contract.md [hunk 4]
```diff
@@ -2,23 +2,35 @@
...
 
-1. Open MetaMask and confirm you are connected to Obscuro Testnet network.
+1. Compile your smart contract using the Remix Solidity Compiler.
+
+1. Open MetaMask and confirm you are connected to Obscuro Testnet network. The parameters for the Obscuro Testnet can be found [here](./essentials.md).
```
Comment (joeldudley) [inline docs/testnet/deploying-a-smart-contract.md:22]: Rather than "Open MetaMask", I'd say "Log in to MetaMask".

File: docs/testnet/deploying-a-smart-contract.md [hunk 5]
```diff
@@ -2,23 +2,35 @@
...
 
 1. In the _Deploy & Run Transactions_ section of Remix change the Environment to _Injected Web3_. This tells Remix to use the network settings currently configured in your MetaMask wallet, which in this case is the Obscuro Testnet.
 
+    If the connection to Obscuro Testnet is successful you will see the text _Custom (777) network_ displayed under _Injected Web3_.
```
Comment (joeldudley) [inline docs/testnet/deploying-a-smart-contract.md:26]: Not sure if this works formatting wise. Check it after deployment if you haven't already.

File: docs/testnet/deploying-a-smart-contract.md [hunk 6]
```diff
@@ -2,23 +2,35 @@
...
+
 1. Click the _Deploy_ button to deploy your smart contract to the Obscuro Testnet.
+
+1. MetaMask will automatically open and ask you to confirm the deployment. Click _Confirm_.
```
Comment (joeldudley) [inline docs/testnet/deploying-a-smart-contract.md:30]: I'd say "confirm the transaction".

I think we need an additional step to say, "Wait for your transaction to be confirmed. Notifications will be shown in MetaMask and Remix to indicate whether the transaction was successful."

File: docs/testnet/deploying-a-smart-contract.md [hunk 7]
```diff
@@ -2,23 +2,35 @@
...
+
+Congratulations, your smart contract is now deployed to Obscuro Testnet!
+
+Because Obscuro provides full privacy the details of any transactions with your smart contract are encrypted and only visible to those with the viewing key.
```
Comment (joeldudley) [inline docs/testnet/deploying-a-smart-contract.md:34]: This isn't quite right. The rules on who can see what are a bit complex. But in particular, any can see transactions they submitted.

So I'd reword it as, "Because Obscuro provides full privacy, the details of your transaction are encrypted and only visible to you, as the holder of your wallet's private key." (it's not like you can share viewing keys around, and the viewing keys are ephemeral, but as long as you have the wallet private key, you can retrieve the transaction details).

File: docs/testnet/deploying-a-smart-contract.md [hunk 8]
```diff
@@ -2,23 +2,35 @@
...
+
+Now head over to the [ObscuroScan page](./obscuroscan.md) to see how you can view the transaction details.
```
Comment (joeldudley) [inline docs/testnet/deploying-a-smart-contract.md:36]: One weakness of ObscuroScan atm is that you can only see the latest txs and rollups flowing through. It would be nice to extend it so that you could grab a specific transaction and view it.

File: docs/testnet/obscuroscan.md [hunk 9]
```diff
@@ -4,15 +4,15 @@
...
+Notice the _Encrypted Transaction Blob_ field. You can copy and paste the field contents into ObscuroScan's _Decrypt rollup transaction blobs (testnet only!)_ page to see the content of the transaction in unencrypted plain text. This is a feature of Testnet. In contrast, on Mainnet, rollups will use rotating keys that are not known to anyone, or anything, other than the Obscuro enclaves.
 
 ## How to Decrypt Transaction Blobs
-Decrypting transaction blobs is only possible on Testnet to help you understand how Obscuro works. ObscuroScan on Testnet uses a rollup encryption key which is long-lived and well-known. On Mainnet, rollups will use rotating keys that are not known to anyone, or anything, other than the Obscuro enclaves.
+Decrypting transaction blobs is only possible on Testnet to help you understand how Obscuro works. ObscuroScan on Testnet uses a rollup encryption key which is long-lived and well-known. In contrast, on Mainnet, rollups will use rotating keys that are not known to anyone, or anything, other than the Obscuro enclaves.
```
Comment (joeldudley) [inline docs/testnet/obscuroscan.md:14]: The sentence starting "In contrast, ..." is repeated from the previous paragraph.

File: docs/testnet/wallet-extension.md [hunk 10]
```diff
@@ -32,7 +32,7 @@ unencrypted. If the data is not particularly sensitive, it can also be run in an
 
     * For other OSes and architectures, see `Compiling the binary`, below
 
-2. Start the wallet extension by running the `wallet_extension` binary with the following flags:
+2. Open a terminal window of command prompt. Start the wallet extension by running the `wallet_extension` binary with the following flags:
 
    ```--nodeRPCHTTPAddress=<Obscuro host RPC HTTP address> --nodeRPCWebsocketAddress=<Obscuro host RPC websocket address>```
```
Comment (joeldudley) [inline docs/testnet/wallet-extension.md:37]: Can we give the testnet defaults here? Or do they rotate too often?

**Review by joeldudley — CHANGES_REQUESTED** [review]
Left some comments :)

File: docs/testnet/deploying-a-smart-contract.md (hunk 1 above)
Comment (GavT) [reply docs/testnet/deploying-a-smart-contract.md:11]: ok

File: docs/testnet/deploying-a-smart-contract.md (hunk 3 above)
Comment (GavT) [reply docs/testnet/deploying-a-smart-contract.md:18]: ok

File: docs/testnet/deploying-a-smart-contract.md (hunk 4 above)
Comment (GavT) [reply docs/testnet/deploying-a-smart-contract.md:22]: ok

File: docs/testnet/deploying-a-smart-contract.md (hunk 6 above)
Comment (GavT) [reply docs/testnet/deploying-a-smart-contract.md:30]: ok

File: docs/testnet/deploying-a-smart-contract.md (hunk 2 above)
Comment (GavT) [reply docs/testnet/deploying-a-smart-contract.md:13]: ok

File: docs/testnet/deploying-a-smart-contract.md (hunk 7 above)
Comment (GavT) [reply docs/testnet/deploying-a-smart-contract.md:34]: ok

File: docs/testnet/deploying-a-smart-contract.md (hunk 8 above)
Comment (GavT) [reply docs/testnet/deploying-a-smart-contract.md:36]: Yeah, that seems like a high priority. The ObscuroScan could soak up a ton of effort so we need to prioritise ruthlessly.

File: docs/testnet/wallet-extension.md (hunk 10 above)
Comment (GavT) [reply docs/testnet/wallet-extension.md:37]: I've flip flopped here myself but actually testnet.obscu.ro and the port numbers are likely to be fairly static. I'll change the text.

File: docs/testnet/deploying-a-smart-contract.md (hunk 5 above)
Comment (GavT) [reply docs/testnet/deploying-a-smart-contract.md:26]: What do you mean?

File: docs/testnet/obscuroscan.md (hunk 9 above)
Comment (GavT) [reply docs/testnet/obscuroscan.md]: ok

File: docs/testnet/deploying-a-smart-contract.md (hunk 5 above)
Comment (joeldudley) [reply docs/testnet/deploying-a-smart-contract.md:28]: Don't you need to make it `    * If the connection` or something?

**Comment by joeldudley** [comment]
@GavT I've realised. We should say something to the effect of:

"All accounts on the Obscuro testnet are automatically prefunded and do not require the user to obtain any funds (e.g. via a faucet)."

File: docs/testnet/deploying-a-smart-contract.md (hunk 5 above)
Comment (GavT) [reply docs/testnet/deploying-a-smart-contract.md:28]: ah ok, fixed

**Comment by joeldudley** [comment]
> > 
> 
> This is in the intro: "As you would expect from a Testnet, OBX tokens issued on Obscuro Testnet have no value. There are no gas fees, for now at least. As Testnet develops gas fees will be introduced along with an OBX faucet."
> 
> But you have introduced gas fees, right?

I can't see that bit. Is it on another page? It might be good to recap it on the deploy page if so.

In practice, what we do is preallocate a bunch of eth to the sending account just before we process the transaction, so it's effectively the same as no gas fees.

**Comment by GavT** [comment]
> > > 
> > 
> > 
> > This is in the intro: "As you would expect from a Testnet, OBX tokens issued on Obscuro Testnet have no value. There are no gas fees, for now at least. As Testnet develops gas fees will be introduced along with an OBX faucet."
> > But you have introduced gas fees, right?
> 
> I can't see that bit. Is it on another page? It might be good to recap it on the deploy page if so.
> 
> In practice, what we do is preallocate a bunch of eth to the sending account just before we process the transaction, so it's effectively the same as no gas fees.

OK. First para updated on the deploy page. Can you take a look @joeldudley

**Review by joeldudley — APPROVED** [review]
Lgtm
